src/Symfony/Bundle/FrameworkBundle/Command/MessageConsumeCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Message\Asynchronous\Transport\ReceivedMessage;
+use Symfony\Component\Message\MessageBusInterface;
+use Symfony\Component\Message\Transport\ReceiverInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessageConsumeCommand extends Command
+{
+    protected static $defaultName = 'message:consume';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('receiver', InputArgument::REQUIRED, 'Name of the receiver'),
+                new InputOption('bus', 'b', InputOption::VALUE_REQUIRED, 'Name of the bus to dispatch the messages to', 'message_bus'),
+            ))
+            ->setDescription('Consume a message')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command consume a message and dispatch it to the message bus.
+
+    %command.full_name% <consumer-service-name>
+
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var ContainerInterface $container */
+        $container = $this->getApplication()->getKernel()->getContainer();
+
+        if (!$container->has($receiverName = $input->getArgument('receiver'))) {
+            throw new \RuntimeException(sprintf('Receiver "%s" do not exists', $receiverName));
+        } elseif (!($receiver = $container->get($receiverName)) instanceof ReceiverInterface) {
+            throw new \RuntimeException(sprintf('Receiver "%s" is not a valid message consumer. It should implement the interface "%s"', $receiverName, ReceiverInterface::class));
+        }
+
+        if (!$container->has($busName = $input->getOption('bus'))) {
+            throw new \RuntimeException(sprintf('Bus "%s" do not exists', $busName));
+        } elseif (!($messageBus = $container->get($busName)) instanceof MessageBusInterface) {
+            throw new \RuntimeException(sprintf('Bus "%s" is not a valid message bus. It should implement the interface "%s"', $busName, MessageBusInterface::class));
+        }
+
+        foreach ($receiver->receive() as $message) {
```

Comment:
This kind of API wouldn't allow gaining the best possible performance while dealing with AMQP and consuming from several queues at the same time.

Reply:
@makasim what is the main blocker in "gaining the best possible performance"? I believe this is up to the adapters anyway? Maybe things we could do on the enqueue adapter for example? 🤔

Reply:
Is receiver able to consume only from one queue? Should I spawn 100 processes (for each queue) if I have 100 queues?

Reply:
Do you in reality have 100 queues? In our biggest project we have 10.

Reply:
I don't see why you couldn't pull from multiple queues at the same time from the adapter(s) ?

Reply:
> Do you in reality have 100 queues? In our biggest project we have 10.

* Let's say we need 50 Mb to boot up an application (load kernel, container etc). So we need 10  processes to serve ten queues, right?  You need at least 500Mb to boot up them and then while working they eat more. 
* CPU is used a bit more.
* In development, it is very convenient to run a single command to consume from all queues. 
* There are high load queues and rarely used ones. You can group rarely used ones to be served by a single command, saved memory could be used to run additional consumers for high load queues.

Reply:
> I don't see why you couldn't pull from multiple queues at the same time from the adapter(s) ?

Let's talk about AMQP. To get the most of it we should use AMQP's [basic.consume](http://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.consume) method. 

Here's AMQP interop based [example](https://gist.github.com/makasim/7a9f91a38fe8547f533f398e2a99375e) through same is applied to php-amqplib, bunny and amqp-ext. 

@sroze How would make the example work with the current receiver interface.